datafusion/sqllogictest/test_files/array_query.slt

```diff
@@ -19,7 +19,7 @@
 
 # Make a table with multiple input partitions
 statement ok
-CREATE TABLE data AS
```

Comment:
Seems there're a file or directory called `data` in the working directory. However, I think it's really weird. IMO, we should only match a string literal like `select * from '/xxx/aaa/data.csv'` but the normal identifier of a table also be matched.
I'll do more research for it.

Reply:
ok. I did an experimental for DataFusion v39.0.0 like
```
(base) jax@jax-canner:~/git/datafusion/testing/data/csv (master~4) $ mkdir datas 
(base) jax@jax-canner:~/git/datafusion/testing/data/csv (master~4) $ cp aggregate_test_100.csv datas 
(base) jax@jax-canner:~/git/datafusion/testing/data/csv (master~4) $ datafusion-cli                 
DataFusion CLI v39.0.0
> select * from datas;
+----------+----------+----------+----------+-------------+----------------------+----------+----------+------------+----------------------+-------------+---------------------+--------------------------------+
| column_1 | column_2 | column_3 | column_4 | column_5    | column_6             | column_7 | column_8 | column_9   | column_10            | column_11   | column_12           | column_13                      |
+----------+----------+----------+----------+-------------+----------------------+----------+----------+------------+----------------------+-------------+---------------------+--------------------------------+

```
I created a folder `datas` and place a csv in this, then start the cli in this path. I can directly query this folder without single-quote string literal. 

For a single file, it doesn't work
```
(base) jax@jax-canner:~/git/datafusion/testing/data/csv (master~4) $ cp aggregate_test_100.csv data
(base) jax@jax-canner:~/git/datafusion/testing/data/csv (master~4) $ datafusion-cli                
DataFusion CLI v39.0.0
> select * from data;
Error during planning: table 'datafusion.public.data' not found
```

I think it should be an issue to match only the string literal as the dynamic file table name.